services/jvm/src/main/scala/io/gearpump/services/MasterService.scala

```diff
@@ -184,49 +198,97 @@ object MasterService {
 
   case class BuiltinPartitioners(partitioners: Array[String])
 
-  case class AppSubmissionResult(success: Boolean, appId: Int)
+  case class AppSubmissionResult(success: Boolean)
 
   case class Status(success: Boolean, reason: String = null)
 
   /**
-   * Upload user application (JAR) into temporary directory and use AppSubmitter to submit
-   * it to master. The temporary file will be removed after submission is done/failed.
+   * Submit Native Application.
+   */
+  def submitGearApp(jar: Option[File], userConf: Option[File], args: String, sysConfig: Config): Boolean = {
+    val mainClass = "io.gearpump.cluster.main.AppSumitter"
+    val fileMap = Map("jar" -> jar, "gear" -> userConf)
+      .filter(_._2.isDefined).mapValues(_.get)
+    submitAndDeleteTempFiles(fileMap, argStringToArray(args), sysConfig, mainClass)
+  }
+
+  /**
+   * Submit Storm application.
    */
-  def submitJar(jar: File, userConf: Option[File], extraArgs: Array[String],
-                sysConfig: Config): Int = {
+  def submitStormApp(jar: Option[File], stormConf: Option[File], args: String, sysConfig: Config): Boolean = {
+    val mainClass = "io.gearpump.experiments.storm.StormRunner"
+    val fileMap = Map("jar" -> jar, "config" -> stormConf)
+      .filter(_._2.isDefined).mapValues(_.get)
+    submitAndDeleteTempFiles(fileMap, argStringToArray(args), sysConfig, mainClass)
+  }
 
+  private def submitAndDeleteTempFiles(fileMap: Map[String, File], argsArray: Array[String],
+                                       sysConfig: Config, mainClass: String): Boolean = {
     try {
-      val masters = sysConfig.getStringList(Constants.GEARPUMP_CLUSTER_MASTERS).toList.flatMap(Util.parseHostList)
-      val mastersOption = masters.zipWithIndex.map { kv =>
-        val (master, index) = kv
-        s"-D${Constants.GEARPUMP_CLUSTER_MASTERS}.$index=${master.host}:${master.port}"
-      }.toArray
-
-      val hostname = sysConfig.getString(Constants.GEARPUMP_HOSTNAME)
-      var options = Array(
-        s"-D${Constants.GEARPUMP_HOSTNAME}=$hostname",
-        s"-D${PREFER_IPV4}=true"
-      ) ++ mastersOption
-
-      if (userConf.isDefined) {
-        options :+= s"-D${Constants.GEARPUMP_CUSTOM_CONFIG_FILE}=${userConf.get.getPath}"
+      val jar = fileMap.get("jar")
+      if (jar.isEmpty) {
+        throw new IOException("JAR file not supplied")
       }
 
-      val arguments = Array("-jar", jar.getPath) ++ extraArgs
-      val mainClass = AppSubmitter.getClass.getName.dropRight(1)
-      val process = Util.startProcess(options, Util.getCurrentClassPath, mainClass, arguments)
+      val stormLibs = Array(System.getProperty("gearpump.home") + "/lib/storm/*")
```

Comment:
1. I think only storm application requires this classpath. 
2. we have a `util.Constants.GEARPUMP_HOME`